### PR TITLE
Turn exercise 025 to practice

### PR DESCRIPTION
**Pull request by Ozyugoo**

### CODE REVIEW

**Comment by Ozyugoo** [comment]
Hi, @cuihtlauac in the file `answer/impl.ml` on line 1, I got this warning when I tried running `dune build --root .`
>dune
> File "answer/impl.ml", line 1, characters 8-19:
1 | let rec permutation list =
            ^^^^^^^^^^^
Error (warning 39 [unused-rec-flag]): unused rec flag.

But when I removed the rec on it, it complied and passed the `dune test --root . answer` test.
Please check it on your end thank you.

File: practice/025/answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  in
+  let extract_rand list len =
+    extract [] (Random.int len) list
+  in
+  let rec aux acc list len =
+    if len = 0 then acc else
+      let picked, rest = extract_rand list len in
+      aux (picked :: acc) rest (len - 1)
+  in
+  aux [] list (List.length list)
```
Comment (cuihtlauac) [inline practice/025/answer/impl.ml]:
```suggestion
  aux [] list (List.length list)

```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Comment by Ozyugoo** [comment]
> Hi, @cuihtlauac in the file `answer/impl.ml` on line 1, I got this warning when I tried running `dune build --root .`
> 
> > dune
> > File "answer/impl.ml", line 1, characters 8-19:
> > 1 | let rec permutation list =
> > ^^^^^^^^^^^
> > Error (warning 39 [unused-rec-flag]): unused rec flag.
> 
> But when I removed the rec on it, it complied and passed the `dune test --root . answer` test. Please check it on your end thank you.

Hi @cuihtlauac, you didn't say anything or the unused-rec-flag, should I remove the unused the rec or is there other options to it?

**Comment by cuihtlauac** [comment]
> Hi @cuihtlauac, you didn't say anything or the unused-rec-flag, should I remove the unused the rec or is there other options to it?

The `rec` keyword is a error in the initial example.
- Remove it from the `practice/answer/impl.ml` file
- Remove it from the `data/exercise/025_generate_permutations.md`

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Yes, the `rec` keyword must be removed here

**Comment by Ozyugoo** [comment]
Hey @cuihtlauac, I removed the rec from exercise 025 and the practice/answer/impl.ml.

File: practice/025/answer/impl.ml
Comment (cuihtlauac) [inline practice/025/answer/impl.ml:1]: Sorry, still the same problem with end of files :-)

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
Thanks @Ozyugoo. I'm merging without your last commit which is useless (PR #2292)

**Pull request closed without merging**